### PR TITLE
Add preloader to calendar and show 404

### PR DESCRIPTION
**Pull request by tiffanyqi**
- add preloader to calendar
- 404 page, of lake victoria basically
- to see the 404, set debug to False in Dev

### CODE REVIEW

File: cal/cal/static/js/components/calendar.js [hunk 1]
```diff
@@ -131,9 +142,11 @@ analyticsApp.controller('CalendarCtrl', function ($scope, $http, $q, uiCalendarC
        firstDay: 1,
        eventRender: $scope.eventRender,
        viewRender: this.viewRender,
+       loading: $scope.loading
      }
    };
-   
+  
+
```
Comment (jamesuejio) [inline cal/cal/static/js/components/calendar.js]: Whats with the two extra lines?

**Review by jamesuejio — CHANGES_REQUESTED** [review]
404 doesn't seem to work for me. http://localhost:8000/fdsfsda gives regular Page Not Found error. Could be bower stuff, so maybe merge master in and lmk again :)

File: cal/cal/static/js/components/calendar.js [hunk 2]
```diff
@@ -108,6 +108,16 @@ analyticsApp.controller('CalendarCtrl', function ($scope, $http, $q, uiCalendarC
      return element;
    };
    
+   $scope.loading = function(isLoading, view){
```
Comment (jamesuejio) [inline cal/cal/static/js/components/calendar.js]: I thought we didn't want to use $scope? @MaxFangX ?

**Review by jamesuejio — CHANGES_REQUESTED** [review]

File: cal/cal/views.py [hunk 3]
```diff
@@ -139,3 +140,9 @@ def google_auth(request):
     else:
         auth_uri = flow.step1_get_authorize_url()
         return HttpResponseRedirect(auth_uri)
+
+def handler404(request):
+    response = render_to_response('404.html', {},
```
Comment (MaxFangX) [inline cal/cal/views.py]: `render_to_response` [is "likely to be deprecated in the future"](https://docs.djangoproject.com/en/1.10/topics/http/shortcuts/#render-to-response)

File: cal/cal/static/js/components/calendar.js (hunk 2 above)
Comment (MaxFangX) [reply cal/cal/static/js/components/calendar.js]: Correct! Bind to `this`

File: cal/cal/templates/404.html [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{% extends "base.html" %}
+{% load staticfiles %}
+
+{% block content %}
+
+<div class="container" style="text-align: center; height: 100%">
+
+    <h2>This link doesn't exist! :( Here's a beautiful waterfront. When you're ready go to back, click <a href="/">here to return to your dashboard</a></h2>.
```
Comment (MaxFangX) [inline cal/cal/templates/404.html]: Our html is 2 spaces, not 4

**Review by MaxFangX — COMMENTED** [review]
Small fixes, but otherwise LGTM

File: cal/cal/static/js/components/calendar.js (hunk 1 above)
Comment (tiffanyqi) [reply cal/cal/static/js/components/calendar.js]: must have been an accident; thanks for checking! fixing

File: cal/cal/templates/404.html (hunk 4 above)
Comment (tiffanyqi) [reply cal/cal/templates/404.html]: cool, fixing

File: cal/cal/static/js/components/calendar.js (hunk 2 above)
Comment (tiffanyqi) [reply cal/cal/static/js/components/calendar.js]: what about like `$scope.eventRender` right above it? or are we going to refactor those things later? changing this anyway, just wondering

**Review by tiffanyqi — COMMENTED** [review]
Looks like some comments disappeared after my push: essentially this is what it was: 
- removed two line breaks in calendar.js
- changed `$scope.loading` to `this.loading` with a `bind(this)`
- changed from 4 spaces to 2 spaces in 404.html

File: cal/cal/views.py (hunk 3 above)
Comment (tiffanyqi) [reply cal/cal/views.py]: @MaxFangX noooo, what should i do instead ;(

**Comment by tiffanyqi** [comment]
@jalagar 404 does it work! If you go to settings.py and change your current env to `Debug = False`, you'll see it

**Comment by tiffanyqi** [comment]
@jalagar implemented @MaxFangX 's changes

**Review by jamesuejio — APPROVED** [review]

**Pull request merged**